### PR TITLE
Added logistic_regression.py and unit testing file

### PR DESCRIPTION
**Pull request by rohit-khoiwal-30**
1. Added Logistic Regression python file. 
2. Added unit testing file for logistic regression.

### CODE REVIEW

File: tests/test_logistic_regression.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    n_f = 5 #no. of features
+    n_c = 3 #no. of classes
+
+    parameters = init_weights(n_f, n_c, random_key= 1)
+
+    assert (n_f,n_c) == parameters["weights"].shape
+    assert (1, n_c) == parameters["bias"].shape
+
+
+def test_lr_fit():
```
Comment (murphyk) [inline tests/test_logistic_regression.py:45]: I suggest you compare your fitting method to sklearn. Since the problem is convex you should get the same answer (using `np.allclose(results_jax, results_sklearn, atol=1e-2)` say)

File: probml_utils/logistic_regression.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        random_key = unique key to generate pseudo random numbers
+    return:
+        parameters: a dictionary contains weights and bias    
+    """
+    parameters = {}
+    parameters["weights"] = jax.random.normal(key = jax.random.PRNGKey(random_key), shape = [n_f,n_c])
+    parameters["bias"] = jnp.zeros((1,n_c))
+    return parameters
+    
+def fit(x, y, max_iter = 1000, learning_rate = 0.75, lambd = 0.001, random_key = 1):
```
Comment (murphyk) [inline probml_utils/logistic_regression.py:55]: A learning rate of 0.75 seems very high...

File: probml_utils/logistic_regression.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    """
+    Arguments:
+        x = training dataset, shape = (no. of examples, no. of features)
+        y = targets, shape = (no. of example, )
+        max_iter = maximum no. of iteration algorithms run
+        learning_rate = stepsize to update the parameters
```
Comment (murphyk) [inline probml_utils/logistic_regression.py:61]: The comment string should say that you use adam (otherwise you need to read the code to figure this out).
I suggest also making a BFGS version.

**Review by murphyk — CHANGES_REQUESTED** [review]

File: probml_utils/logistic_regression.py (hunk 2 above)
Comment (rohit-khoiwal-30) [reply probml_utils/logistic_regression.py]: update learning rate to 0.1

File: probml_utils/logistic_regression.py (hunk 3 above)
Comment (rohit-khoiwal-30) [reply probml_utils/logistic_regression.py:67]: I have added a comment for adam optimizer and also implemented the BFGS method but I have to generalize the algorithm for the multi-class problems.

**Comment by rohit-khoiwal-30** [comment]
I have added the LBFGS method from jaxopt library. 
Update the unit_test file.

 @murphyk  @nipunbatra @karm-patel @patel-zeel
